uptime/checker.go

```diff
@@ -0,0 +1,416 @@
+package uptime
+
+import (
+	"context"
+	"sync"
+	"fmt"
+	"time"
+
+	logging "github.com/ipfs/go-log/v2"
+
+	"github.com/filecoin-project/lotus/api/v0api"
+	"github.com/filecoin-project/go-address"
+	chainTypes "github.com/filecoin-project/lotus/chain/types"
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/go-state-types/big"
+
+	"github.com/libp2p/go-libp2p/p2p/protocol/ping"
+	"github.com/libp2p/go-libp2p-core/host"
+	libp2pMultiaddr "github.com/multiformats/go-multiaddr"
+	peerstore "github.com/libp2p/go-libp2p-core/peer"
+)
+
+var log = logging.Logger("uptime")
+
+const NEW_CHECKER = 2
+const REPORT_CHECKER = 8
+
+// UptimeChecker maintains the uptime of member nodes
+type UptimeChecker struct {
+	api v0api.FullNode
+
+	self ActorID
+	actorAddress address.Address
+	
+	checkerAddresses []MultiAddr
+	nodeAddresses map[ActorID]map[MultiAddr]HealtcheckInfo
+
+	// libp2p ping related
+	node host.Host
+	ping *ping.PingService
+
+	rwLock sync.RWMutex
+	stop bool
+}
+
+func NewUptimeChecker(
+	api v0api.FullNode,
+	actorAddress string,
+	checkerAddresses []MultiAddr,
+	self ActorID,
+	node host.Host,
+	ping *ping.PingService,
+) (UptimeChecker, error) {
+	addr, err := address.NewFromString(actorAddress)
```

Comment:
It is not clear if this is the actor address of the uptime checker, or our own wallet address. The same happens with `ActorID`.